versioned/persist/tests/src/main/java/org/projectnessie/versioned/persist/tests/AbstractMergeTransplant.java

```diff
@@ -180,7 +204,25 @@ void apply(
 
       try (Stream<CommitLogEntry> targetLog =
           databaseAdapter.commitLog(databaseAdapter.hashOnReference(target, Optional.empty()))) {
-        assertThat(targetLog).hasSize(i + 1);
+        List<CommitLogEntry> targetLogEntries = targetLog.collect(Collectors.toList());
```

Comment:
The added tests do assert the operations in the commits, which is nice, but not what _should_ be tested.
The keys and contents for every merged/transplanted commit via DatabaseAdapter.keys() + DatabaseAdapter.values() should be tested - that's what your change's touching.

Reply:
I added some validations for keys but I am finding hard to add it in the current testcase. I find easy to write my own testcase for this :(